src/promptflow/tests/sdk_cli_test/e2etests/test_flow_run.py

```diff
@@ -127,6 +129,72 @@ def test_basic_run_bulk(self, azure_open_ai_connection: AzureOpenAIConnection, l
         # write to user_dir/.promptflow/.runs
         assert ".promptflow" in run.properties["output_path"]
 
+    def test_local_storage_delete(self, pf):
+        result = pf.run(flow=f"{FLOWS_DIR}/print_env_var", data=f"{DATAS_DIR}/env_var_names.jsonl")
+        local_storage = LocalStorageOperations(result)
+        local_storage.delete()
+        assert not os.path.exists(local_storage._outputs_path)
+
+    def test_flow_run_delete(self, pf):
+        result = pf.run(flow=f"{FLOWS_DIR}/print_env_var", data=f"{DATAS_DIR}/env_var_names.jsonl")
+        local_storage = LocalStorageOperations(result)
+        output_path = local_storage.path
+
+        # delete new created run by name
+        pf.runs.delete(result.name)
+
+        # check folders and dbs are deleted
+        assert not os.path.exists(output_path)
+
+        from promptflow._sdk._orm import RunInfo as ORMRun
+
+        pytest.raises(RunNotFoundError, lambda: ORMRun.get(result.name))
+        pytest.raises(RunNotFoundError, lambda: pf.runs.get(result.name))
+
+    def test_flow_run_delete_fake_id_raise(self, pf: PFClient):
+        run = "fake_run_id"
+
+        # delete new created run by name
+        pytest.raises(RunNotFoundError, lambda: pf.runs.delete(name=run))
+
+    pytest.mark.skipif(sys.platform == "win32", reason="Windows doesn't support chmod, just test permission errors")
+
+    def test_flow_run_delete_invalid_permission_raise(self, pf: PFClient):
+        result = pf.run(flow=f"{FLOWS_DIR}/print_env_var", data=f"{DATAS_DIR}/env_var_names.jsonl")
+        local_storage = LocalStorageOperations(result)
+        output_path = local_storage.path
+        os.chmod(output_path, 0o555)
+
+        # delete new created run by name
+        pytest.raises(InvalidRunError, lambda: pf.runs.delete(name=result.name))
+
+        # Change folder permission back
+        os.chmod(output_path, 0o755)
+        pf.runs.delete(name=result.name)
+        assert not os.path.exists(output_path)
+
+    def test_flow_run_delete_eval_run(self, pf: PFClient):
```

Comment:
this test name is not that accurate, maybe `test_visualize_run_with_referenced_run_deleted`? or some name else.

Reply:
OK